dex/src/matching.rs

```diff
@@ -869,26 +890,28 @@ impl<'ob> OrderBookState<'ob> {
             _ => (0, 0),
         };
 
-        let out = {
-            let native_qty_still_locked = pc_qty_to_keep_locked * pc_lot_size;
-            let native_qty_unlocked = native_pc_qty_remaining - native_qty_still_locked;
+        if !is_send_take {
+            let out = {
+                let native_qty_still_locked = pc_qty_to_keep_locked * pc_lot_size;
+                let native_qty_unlocked = native_pc_qty_remaining - native_qty_still_locked;
 
-            to_release.unlock_native_pc(native_qty_unlocked);
+                to_release.unlock_native_pc(native_qty_unlocked);
 
-            Event::new(EventView::Out {
-                side: Side::Bid,
-                release_funds: false,
-                native_qty_unlocked,
-                native_qty_still_locked,
-                order_id,
-                owner,
-                owner_slot,
-                client_order_id: NonZeroU64::new(client_order_id),
-            })
-        };
-        event_q
-            .push_back(out)
-            .map_err(|_| DexErrorCode::EventQueueFull)?;
+                Event::new(EventView::Out {
+                    side: Side::Bid,
+                    release_funds: false,
+                    native_qty_unlocked,
+                    native_qty_still_locked,
+                    order_id,
+                    owner,
+                    owner_slot,
+                    client_order_id: NonZeroU64::new(client_order_id),
+                })
+            };
+            event_q
+                .push_back(out)
+                .map_err(|_| DexErrorCode::EventQueueFull)?;
+        }
 
         if pc_qty_to_keep_locked > 0 {
             let bids = self.orders_mut(Side::Bid);
```

Comment:
the other side has a `check_assert!(!is_send_take)?;` here to guard against putting an order on the book accidentally (even if post_allowed should just always be false). Maybe nice to `assert!(!(post_allowed && send_take))` further up.